### PR TITLE
feat(globals): add grouped header nav and fixed footer columns

### PR DESCRIPTION
**Pull request by SebastianSchuetze**
Summary:
This PR introduces grouped header navigation and a structured three-column footer model, including schema, seed, migration, and frontend consumption updates. It also adds targeted unit coverage for new header group-link behavior after review feedback.

Changes:
- Add header link-group support (`type: 'group'`) in field/config and navigation rendering/normalization (`src/fields/link.ts`, `src/globals/Header/config.ts`, `src/utilities/normalizeNavItems.ts`, `src/components/templates/Header/Nav/index.tsx`).
- Refactor footer globals from `navItems` to `aboutLinks`, `serviceLinks`, and `informationLinks`, and update layout/template/story consumers (`src/globals/Footer/config.ts`, `src/app/(frontend)/layout.tsx`, `src/components/templates/Footer/Component.tsx`, `src/stories/templates/Footer.stories.tsx`, `src/stories/templates/fixtures.ts`).
- Update baseline global seed payload and generated Payload types for the new footer/header model (`src/endpoints/seed/data/baseline/globals.json`, `src/endpoints/seed/globals/globals-seed.ts`, `src/payload-types.ts`).
- Add migrations for header enum extension and footer schema/data migration (`src/migrations/20260213_230000_header_nav_group_link_type.ts`, `src/migrations/20260214_000000_footer_fixed_groups.ts`, `src/migrations/index.ts`).
- Add unit tests for header `group` edge cases in `normalizeHeaderNavItems` (`tests/unit/utilities/normalizeNavItems.test.ts`).

Why:
The header now needs label-only grouped entries with submenus, and the footer needs explicit semantic columns instead of slicing unrelated nav data. This change makes navigation configuration deterministic in CMS, removes implicit header/footer coupling, and adds tests for the new group-link behavior to prevent regressions.

Testing:
- `pnpm check`
- `pnpm build`
- `pnpm format`
- `pnpm tests --project=unit --run tests/unit/utilities/normalizeNavItems.test.ts`

Related:

Breaking changes:
- Footer global schema changed from `navItems` to `aboutLinks`, `serviceLinks`, and `informationLinks`.
- Includes database migration to transform existing footer rows into the new structure.

### CODE REVIEW

File: src/components/templates/Footer/Component.tsx [hunk 1]
```diff
@@ -21,44 +21,20 @@ export const Footer: React.FC<FooterProps> = ({ footerNavItems, headerNavItems }
...
-                  </ul>
-                </div>
+                {footerGroups.map((group) => (
+                  <div key={group.title} className="flex flex-col items-start gap-6 pt-6 pl-1.5 md:flex-1 md:basis-0">
+                    <Heading as="h4" size="h6" align="left" className="text-lg text-foreground">
+                      {group.title}
+                    </Heading>
+                    <ul className="space-y-6">
+                      {group.items.map((link) => (
+                        <li key={`${group.title}-${link.href}-${link.label ?? ''}`}>
```
[the remark this answers is not in the document]
Comment (SebastianSchuetze) [reply src/components/templates/Footer/Component.tsx]: I am not changing this in this PR. The three group keys are fixed and deterministic (About, Service, Information), and the current data model does not produce duplicate href plus label pairs per group. Index based keys would be less stable on reordering. If duplicate items become a real case, I will switch to Payload row ids in a dedicated follow up.

File: src/app/(frontend)/layout.tsx [hunk 2]
```diff
@@ -13,16 +13,15 @@ import { draftMode } from 'next/headers'
...
-import { normalizeNavItems, normalizeHeaderNavItems } from '@/utilities/normalizeNavItems'
+import { normalizeFooterNavGroups, normalizeHeaderNavItems } from '@/utilities/normalizeNavItems'
 import type { Footer as FooterType, Header as HeaderType } from '@/payload-types'
 
 export default async function RootLayout({ children }: { children: React.ReactNode }) {
   const { isEnabled } = await draftMode()
-  const footerData: FooterType = await getCachedGlobal('footer', 1)()
-  const headerData: HeaderType = await getCachedGlobal('header', 1)()
+  const footerData = (await getCachedGlobal('footer', 1)()) as FooterType
+  const headerData = (await getCachedGlobal('header', 1)()) as HeaderType
```
[the remark this answers is not in the document]
Comment (SebastianSchuetze) [reply src/app/(frontend)/layout.tsx]: Good point in general. In this file the assertion is currently tied to getCachedGlobal being slug agnostic and returning a broad union, so replacing it with annotation alone does not improve type safety. I am keeping this as is for now and will handle it in a focused typing refactor for getGlobal and getCachedGlobal.

File: src/utilities/normalizeNavItems.ts [hunk 3]
```diff
@@ -68,10 +117,19 @@ export function normalizeHeaderNavItems(
 ): HeaderNavItem[] {
   return (data?.navItems ?? [])
     .map((item) => {
-      const resolved = normalizeLinkRecord(item?.link)
+      const resolved = normalizeLinkRecord(item?.link, { allowGroupWithoutHref: true })
```
[the remark this answers is not in the document]
Comment (SebastianSchuetze) [reply src/utilities/normalizeNavItems.ts]: Implemented in commit dc72199. I added unit tests for all three requested cases: group item without href but with valid sub items, group item without href and without sub items being filtered out, and group links forcing newTab to false.
